### DISCUSSION TITLE
Smart socket as Watchdog with Rules, but something going wrong

### DISCUSSION THREAD

**Discussion by palmtop** [arendst/Tasmota#21548] [General]
I have a remote OrangePi computer which is supposed to regulate the heating in our weekend house, but somehow it is very unstable. So I decided to add a smart switch, flashed with Tasmota (https://tasmota.github.io/docs/devices/OBI-Socket-2/) to act as a Watchdog and if the Pi is not reseting its timer, it simply disconnects the mains from the Pi for 60 seconds and then switches it back. 

Normally it is working well, but after 10 or so days, the smart switch simply switches off the power, and I loose completly the connection. The smart switch gets its wifi from the Pi, so whenever there is a restart it looses WiFi connection, but I assume this should not be a problem.

To make it even more robust, last time I added one more rule, that every day at 00:10 the switch should powen on, but it seems that this doesn't help.

I have the following rules:

```
on Rules#Timer=1 do Backlog RuleTimer2 60; Power Off; ENDON
ON Rules#Timer=2 DO Backlog RuleTimer1 10800; Power On Endon
ON System#Boot DO RuleTimer1 10800; ENDON
ON System#Boot DO Power On ENDON
ON Time#Minute=10 Do Power On ENDON
```

Timer 1 is for a 3 hours watchdog time, Timer 2 is for the 60 sec switch off. On the pi I call http://10.42.0.100/cm?cmnd=RuleTimer1%2010800 in order to reset the watchdog.

Any suggestion what I am missing, what could go wrong?

**Comment by Noschvie** [comment]
Have you checked if there is a watchdog feature as part of the OS of the OrangePi ?

**Reply by barbudor** [reply]
Don't use a timer for the switch off duration
Use:
```
Pulsetime 160
Poweronstate 5
```
These are 1 time config command

**Comment by palmtop** [comment]
@Noschvie: I have enabled the watchdog on the Orange Pi, but it was not reliable, it did not restart it, when needed, so I switched it off by now.

@barbudor: Thank you for the hint, I will try that one, most probably it will be the solution. But I need this not only when the socket is restartaring, but also when I need to restart my Pi.

In the meantime I have found this:  #19719  , I think that it is related, when I made a rule to start a timer when the relay gets powered off and switch it back after 60 second, the power was not reconnected and when I powered it back manually, the log showed that it elapsed only 40 seconds, till I powered on manually, while in the reality it was arround 10 minutes.

And I found this: #21307 which mentions: TimedPower.

I will change the rule to do the restart with the TimedPower and use Poweronstate to set it to ON after botting the Smart Switch, and I hope, that this way it will work.

Unfortunately we just traveled back from the location, and again I can not access it remotely, so I have to wait for the next time we go there and do the changes locally.

**Reply by barbudor** [reply]
You didn't understood my recommendation to use `PowerOnState 5` which works with `PulseTime`
`PowerOnState 5` is not ONLY about getting the relay ON at restart but also to behave as Inverted-PulseTime
- relay default as ON
- relay turns OFF on `Power Off` command and automatically revert to ON after the duration of PulseTime

If you are using TimedPower, then you should use `PowerOnState 1`

**Comment by palmtop** [comment]
Finally I got to the device and did some tests.

I found 4 ways to do a powercycle for the connected device:

1. Using a timer and switching on and off the power
2. Using the blink possibility
3. Using the PulseTime setting
4. Using the TimedPower

Here are the test results:

**Timer**

I did the following rule:

`on event#powercycle do backlog RuleTimer2 60; power off endon on rules#timer=2 do power on endon`

Then issued "Event powercycle" from the web console.

Surprisingly it worked as expected, switched of the power then after 60 seconds switched it back. Last week this was not working...

The reason could be that meanwhile it turned out that the SIM card has expired and the system (both the smartswitch and the PI) was without network and ntp for long time, may be that caused the smart switch to loose its sense of time completly. Before doing these test I have fixed the SIM card.
I tried to reproduce the previous behavior, by disabling internet access, then powering down the plug and plugging it back, but it worked again. Anyway it is prooven that in long therm the timer based solution is not stable.

**Blink**

I have set the bling settings to do one blink and it should last 60 sec:

```
BlinkCount 1
BlinkTime 600
```

Then I have added the following rule:

`on event#powercycle do power blink endon`

This is working, but the led remains lit even, when the relay was disconnected.

**PulseTime**

I have set the pulse time to 60 seconds

`PulseTime1 160`

Then used the following rule:

`on event#powercycle do power off endon`

This was not working, it seems tha PulseTime works only if you switch the power to ON and not when you switch it to OFF. I could experiance this, because when I manually switched back the power, after 60 secons it was disconnected.

**TimedPower**

Here I did not needed to do any prevoius setting, just use the following rule:

`on event#powercycle do timedpower 60000,OFF endon`

This was also working. 

**Final solution**

After running this tests I decided to use theTimedPower approach.

I set that on starting the power should be always on (as @barbudor mentioned in his previous comment):

`PowerOnState 1 `

Then I added the following rule to have a 3 hour watchdog:

```
ON Rules#Timer=1 DO Backlog TimedPower 60000,Off; RuleTimer1 10800; ENDON 
ON System#Init DO RuleTimer1 10800; ENDON
```

Now I will leave it here for some weeks and see if it is working. It is a pitty that the timer based approach worked, because now I can not be sure that the other approaches will work or if there will be a problem with the time they stop also working.

Big thanks again for the comments, it helped me to go in the right direction, and as I got the soultion, I will close this discussion.

**Reply by barbudor** [reply]
For Pulsetime to work in that use case, I wrote twice that you need to use 'Poweronstate 5'
